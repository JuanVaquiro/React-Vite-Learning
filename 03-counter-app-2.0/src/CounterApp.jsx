import PropTypes from 'prop-types'
import { useState } from 'react'



const CounterApp = ({ value }) => {
  const [counter, setCount] = useState(value);

  const Incremet = () => {
    // setCount( counter +1 )
    setCount((c) => c + 1);
  };
  
  const Decrement = () => setCount((c) => c - 1);

  const Reset = () => setCount(value);

  return (
    <div className="container">
      <h1>CounterApp</h1>
      <h2 data-testid='counter' >{counter}</h2>
      <div>
        <button onClick={Incremet}>+1</button>
        <button onClick={Decrement}>-1</button>
        <button aria-label='btn-reset' onClick={Reset}>Reset</button>
      </div>
    </div>
  );
};

CounterApp.propTypes = {    
    value: PropTypes.number.isRequired,
}

export default CounterApp 