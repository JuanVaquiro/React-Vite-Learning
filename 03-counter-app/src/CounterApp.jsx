import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types'

// [rafce} para hacer el esqueleo 

const CounterApp = ( { value } ) => {

    // useState retorna una arreglo con dos valores
    const [ counter, setCounter ] = useState(0);


    //HandAdd
    const handleAdd = () => {
        // return () => console.log('hola mundo');
        // setCounter( counter + 1 );
        setCounter( (c) => c +1)
    };
    

    return (    
        <Fragment>  
            <h1>CounterApp</h1>
            <h2> { counter }</h2>
            <button onClick={  handleAdd } >+1</button>
            {/* <button onClick={ (e) => { handleAdd(e) } }>+1</button> */}
        </Fragment>
      );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};

export default CounterApp;