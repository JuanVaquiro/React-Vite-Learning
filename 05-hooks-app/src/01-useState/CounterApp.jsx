import { useState } from "react";

const CounterApp = () => {
  const [counter, setCounter] = useState({
    Number1: 10,
    Number2: 20,
    Number3: 30,
  });

  const { Number1, Number2, Number3 } = counter;

  const fntCounter = () => {
    setCounter({
      ...counter,
      Number1: Number1 + 1,
    });
  };

  return (
    <div>
      <h1>Counter: {Number1}</h1>
      <h1>Counter: {Number2}</h1>
      <h1>Counter: {Number3}</h1>
      <hr />
      <button onClick={fntCounter}> +1 </button>
    </div>
  );
};

export default CounterApp;
