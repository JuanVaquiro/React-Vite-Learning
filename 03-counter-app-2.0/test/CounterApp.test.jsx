import { render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";

describe('testing el componente <CoumterApp />', () => {
    
    const value = 0

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( <CounterApp value={value} />)
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar el valor incial de 100 ', () => {
        
        render ( <CounterApp value={value} />);

        //
        expect( screen.getByText( 0 ) ).toBeTruthy();

        //
        expect( screen.getByRole("heading",{ level: 2 }).innerHTML ).toContain('0');
    });
}); 