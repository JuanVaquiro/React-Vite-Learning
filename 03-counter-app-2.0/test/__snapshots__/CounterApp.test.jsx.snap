// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing el componente <CoumterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      CounterApp
    </h1>
    <h2>
      0
    </h2>
    <div>
      <button>
        +1
      </button>
      <button>
        -1
      </button>
      <button>
        Reset
      </button>
    </div>
  </div>
</div>
`;
